src/libpgmoneta/walfile/wal_reader.c

```diff
@@ -138,6 +143,9 @@ read_all_page_headers(FILE* file, struct xlog_long_page_header_data* long_header
       pgmoneta_deque_add(wal_file->page_headers, NULL, (uintptr_t) page_header, ValueRef);
       page_number++;
    }
+error:
```

Comment:
Doesn't look right - there should be a `return;` before that

Reply:
before the `error:` label?